lib/ansible/modules/packaging/os/snap.py

```diff
@@ -0,0 +1,189 @@
+'''
+
+RETURN = '''
+msg:
+    description: An informative message on the task's output
+    type: str
+    returned: on error
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def clean_err(err):
+    # Remove whitespaces and newlines that the snap CLI outputs.
+    return err.replace('\n', '').replace('      ', '').replace('"', '\'')
+
+
+def snap_exists(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s info %s" % (snap_path, snap_name)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc, out, err
+
+
+def is_snap_installed(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s list %s" % (snap_path, snap_name)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc, out, err
+
+
+def install_snap(module, snap_name):
+    rc, out, err = is_snap_installed(module, snap_name)
+    if rc == 0:
+        # Snap is already installed
+        return False
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    classic = '--classic' if module.params['classic'] else ''
+
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s install %s %s" % (snap_path, snap_name, classic)
+
+    # Actually install the snap
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    if rc == 0:
+        # Snap has been installed
+        return True
+    else:
+        # Something went wrong
+        err = clean_err(err)
+        module.fail_json(msg=err)
+
+
+def remove_snap(module, snap_name):
+    rc, out, err = is_snap_installed(module, snap_name)
+    if rc != 0:
+        # Snap is not installed
+        return False
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s remove %s" % (snap_path, snap_name)
+
+    # Actually remove the snap
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    if rc == 0:
+        # Snap has been removed
+        return True
+    else:
+        # Something went wrong
+        err = clean_err(err)
+        module.fail_json(msg=err)
+
+
+def main():
+    module_args = dict(
+        name=dict(type='list', required=True),
+        state=dict(type='str', required=False, default='present', choices=['absent', 'present']),
+        classic=dict(type='bool', required=False, default=False)
+    )
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True,
+    )
+    snap_names = module.params['name']
+    state = module.params['state']
+
+    # Check if snaps are valid
+    for snap_name in snap_names:
+        rc, out, err = snap_exists(module, snap_name)
+        if rc != 0:
+            module.fail_json(msg="No snap matching '%s' available" % str(snap_name))
+
+    # Apply changes to the snaps
+    exit_changed = False
+    for snap_name in snap_names:
+        if state == 'present':
+            r_changed = install_snap(module, snap_name)
```

Comment:
also, you could've done it via:
```python
exit_changed |= install_snap(module, snap_name)
```

No need to do that `# If we modified one of the snaps` check afterwards + no need for an extra var.